### ISSUE TITLE
gpu-platform ?

### ISSUE THREAD

**Issue by heroes1412** [zawawawa/gatelessgate#5]
What is this option mean?
--gpu-platform <arg> Select OpenCL platform ID to use for GPU mining (default: -1)

What is OpenCL platform ID ?

**Comment by bikkits** [comment]
paltform ID you can think of as a 'team' of GPU's.

I run a RX480 and an old Nvidia GTX 960 - my AMD card - the RX480 is in the top PCI-E slot in my motherboard - so my computer sees that card as Team 1.

My GTX960 runs secondary, and is installed in the BOTTOM PCI-E slot of my motherboard - my computer sees that as Team 2.

If you have just one RX480 (or any card), then it is either --gpu-platform 0 or --gpu-platform 1. Because if only one card - one team, right?

default: -1 I imagine is how the miner tries to identify your graphics cards - and in which Team they are in - people with the big mining rigs you see pics of with (for example) 3 x identical RX480 cards... And below those three cards there is one slot left and it is an old GTX 960. Your Computer is going to assume that the three nice RX480's are a team (PLATFORM 0, or 1) - and the lonely little Nvidia card, yep, Team 2 (PLATFORM 1).


TLDR:

experiment with --gpu-platform 0, and nothing else - if you get an ERROR that asks you to _choose an OPENCL PLATFORM_  - THEN - you have the correct platform.

So - you choose a team: platformid 0
then you choose the players that you want to use (DEVICE ID)

For example - I have 1x AMD and 3x Nvidia - AMD is platform 0.

Nvidia is platform 1.

I would do:

--gpu-platform 0 (AMD card location, top PCI-E slot)
--device-id 1 (the AMD card in platform 0)

if I had three Nvidia cards underneath those, all lined upn I'd write:

--gpu-platform 1 (Team nVidia/PCI-E slot #2)
--device-id 0 1 2 (graphics card 1, 2 and 3)

- you must specify a team, and then a player. (platform, then a deviceid)

**Comment by bikkits** [comment]
whatever command or .exe file you use to run the miner - it is much easier if you just type that, then ' -h' after it.

Somewhere above or below the part that shows you platform commands there should also be something like --gpu-detect-platform-id or device-id etc.

If all else fails just start with platform 0, device 0.
If not work, platform 0 device 1
if not work, platform1 device 0
if not work, platform 0 device 0

- you will wil eventually... 

I do not use this miner, sorry I can't be of more help.

**Comment by heroes1412** [comment]
thanks, i understood.

it seems, gatelessgate is not optimum than claymore tool :(. Hashrate is lower.

**Comment by bikkits** [comment]
mmm - every miner has its ups and downs - I like what is simple, works well and is not complicated. Pool mining has been made very easy by these generous dudes who develop the miners for us - but often once you fully understand Claymore then you can have that .bat file safely tucked away - knowing that it will work every time.

Other mining developments offer more reliable/consistent/configurable/potential if you fully understand how the code works - which I hardly do... I run EBWF CUDA ZEC miner from a one line .bat file to start my GTX 960 - then I have my RX480 setup using the config.txt in Claymore's ZEC GPU miner - I find it easier that way.

As long as you keep your fans/temps steady and you dont get any crashes then we're golden.. really! :p

**Comment by heroes1412** [comment]
Your job is great, keep developing 👍 👍 👍 👍 👍 
Do you have a central manager tool for gatelessgate?

**Comment by bikkits** [comment]
haha this is not my miner! I just subscribed to this.

**Comment by bikkits** [comment]
if you are mining to a pool then there are some good options - yeah - we should keep this space free so it doesn't evolve too much off topic, though :P
